### PR TITLE
Use `getContentFromURLMiddleware` everywhere

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What?
Replaces the similar but different logic that was being used on the `GET /Article` route for the prod build with the new `getContentFromURLMiddleware` middleware that [was added to](https://github.com/guardian/dotcom-rendering/pull/3957) the same route for the dev build.

## Why?
Because [we improved](https://github.com/guardian/dotcom-rendering/pull/4037) `getContentFromURLMiddleware` to handle query params better and we want that same improved goodness when we run `make run-ci`

### CODE REVIEW

**Review by mxdvl — APPROVED** [review]
Seems sound. Could you point to the original PR implementing `getContentFromURLMiddleware`?

File: dotcom-rendering/src/server/prod-server.ts [hunk 1]
```diff
@@ -87,34 +73,26 @@ export const prodServer = () => {
...
-			req.body = config;
-			return renderArticle(req, res);
-		} catch (error) {
-			// eslint-disable-next-line no-console
-			console.error(error);
-		}
-	});
+	app.get(
+		'/Article',
+		[logRenderTime, getContentFromURLMiddleware],
```
Comment (OllysCoding) [inline dotcom-rendering/src/server/prod-server.ts:78]: Should this be in an array, could it be `('/Article', logRenderTime, getContentFromURLMiddleware, async () ...)`? I don't know if there's a difference, I just haven't seen array middleware syntax in express before

**Review by OllysCoding — APPROVED** [review]

File: dotcom-rendering/src/server/prod-server.ts (hunk 1 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/server/prod-server.ts:78]: The [express docs](https://expressjs.com/en/guide/using-middleware.html#middleware.application) suggest it should be an array.

Their example:

```js
const logStuff = [logOriginalUrl, logMethod]
app.get('/user/:id', logStuff, (req, res, next) => {
  res.send('User Info')
})
```

**Comment by oliverlloyd** [comment]
> Seems sound. Could you point to the original PR implementing `getContentFromURLMiddleware`?

Great idea. I've added links to both the original PR and the one where it was improved

File: dotcom-rendering/src/server/prod-server.ts (hunk 1 above)
Comment (jamesgorrie) [reply dotcom-rendering/src/server/prod-server.ts:78]: Taking the line from the docs:

> Middleware can also be declared in an array for reusability.

Interestingly we've not made that abstraction (and I don't think we should), that just makes me think it's not recommended, just useful, as both ways are used in the docs.

Just a wondering though.

**Review by jamesgorrie — APPROVED** [review]
Nice and simple. As these methods aren't tested (as I can see) are we _sure_ they're identical as we're touching `prod-server` here?

File: dotcom-rendering/src/server/prod-server.ts (hunk 1 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/server/prod-server.ts]: I think you're both right, it's nicer without the array. Done!

**Comment by oliverlloyd** [comment]
> Nice and simple. As these methods aren't tested (as I can see) are we _sure_ they're identical as we're touching `prod-server` here?

_Technically_ this is prod code but really it's dev code. There was actually [a comment in to code](https://github.com/guardian/dotcom-rendering/blob/c27c5b85e24913bb0770488baf75e1bb696616ad/dotcom-rendering/src/server/prod-server.ts#L28) that made reference to how they (dev and prod local endpoints) are doing the same thing

**Comment by jamesgorrie** [comment]
> Technically this is prod code but really it's dev code.

Little confused on this - these endpoints get hit by frontend don't they?

**Comment by oliverlloyd** [comment]
> > Technically this is prod code but really it's dev code.
> 
> Little confused on this - these endpoints get hit by frontend don't they?

No. `GET /Article` is purely for local development and testing. It's a convenience endpoint so that we don't need to make POST requests from our local machines. The actual prod server only expects POST calls from Frontend

**Comment by jamesgorrie** [comment]
> No. `GET /Article` is purely for local development and testing. It's a convenience endpoint so that we don't need to make POST requests from our local machines. The actual prod server only expects POST calls from Frontend

Crumbs - took me a while to work that out. Apologies.

**Comment by oliverlloyd** [comment]
> > No. `GET /Article` is purely for local development and testing. It's a convenience endpoint so that we don't need to make POST requests from our local machines. The actual prod server only expects POST calls from Frontend
> 
> Crumbs - took me a while to work that out. Apologies.

To be fair, I think it took me 4 months to get my head around the DCR isn't a web server thing

**Pull request merged**